### PR TITLE
DOC add example to scipy.special.hermite

### PR DESCRIPTION
**Pull request by ghost**
Add examples showcasing scipy `special.hermite()` function for a polynomial of degree 3, with monic set to options True and False. 

```
Examples
    --------
    >>> from scipy import special
    >>> import matplotlib.pyplot as plt
    >>> import numpy as np
    
    >>> p_monic = special.hermite(3, monic=True)
    >>> p_monic
    poly1d([ 1. ,  0. , -1.5,  0. ])
    >>> p_monic(1) # Evaluate to a point
    -0.49999999999999983
    >>> x = np.linspace(-3,3,400) # Choose x in (-3,3)
    >>> y = p_monic(x) # Evaluate p_monic to x
    >>> plt.plot(x,y) # Generate plot
    >>> plt.title("Monic Hermite polynomial of degree 3")
    >>> plt.xlabel("x")
    >>> plt.ylabel("H_3(x)")
    >>> plt.show()
```

#### Reference issue
 Contributes to #7168

#### What does this implement/fix?
 Add example section to docstring

### CODE REVIEW

File: scipy/special/orthogonal.py [hunk 1]
```diff
@@ -1143,6 +1143,25 @@ def hermite(n, monic=False):
...
+    Examples
+    --------
+    >>> from scipy import special
+    >>> import matplotlib.pyplot as plt
+    >>> import numpy as np
+
+    >>> p_monic = special.hermite(3, monic=True)
+    >>> p_monic
+    poly1d([ 1. ,  0. , -1.5,  0. ])
+    >>> p_monic(1) # Evaluate to a point
```
Comment (WarrenWeckesser) [inline scipy/special/orthogonal.py:1155]: Nitpick: we don't really need this comment and the other comments below.  They don't add information that is not already clear from the statement itself.

File: scipy/special/orthogonal.py [hunk 2]
```diff
@@ -1143,6 +1143,25 @@ def hermite(n, monic=False):
...
+    -0.49999999999999983
+    >>> x = np.linspace(-3,3,400) # Choose x in (-3,3)
```
Comment (WarrenWeckesser) [inline scipy/special/orthogonal.py:1157]: [PEP-8](https://www.python.org/dev/peps/pep-0008/): We need spaces after the commas:
```suggestion
    >>> x = np.linspace(-3, 3, 400)
```

**Review by WarrenWeckesser — CHANGES_REQUESTED** [review]
Looks good (although I haven't built the docs locally yet).  I suggested a couple small changes.

**Comment by person142** [comment]
I'll leave the review here to @WarrenWeckesser.

File: scipy/special/orthogonal.py (hunk 1 above)
Comment (lgfunderburk) [reply scipy/special/orthogonal.py]: Removed comments

**Comment by WarrenWeckesser** [comment]
Quick link to the rendered docstring: https://31036-1460385-gh.circle-artifacts.com/0/html-scipyorg/reference/generated/scipy.special.hermite.html#scipy.special.hermite

File: scipy/special/orthogonal.py [hunk 3]
```diff
@@ -1143,6 +1143,25 @@ def hermite(n, monic=False):
...
+    >>> import numpy as np
+
+    >>> p_monic = special.hermite(3, monic=True)
+    >>> p_monic
+    poly1d([ 1. ,  0. , -1.5,  0. ])
+    >>> p_monic(1)
+    -0.49999999999999983
+    >>> x = np.linspace(-3, 3, 400)
+    >>> y = p_monic(x)
+    >>> plt.plot(x,y)
```
Comment (WarrenWeckesser) [inline scipy/special/orthogonal.py:1159]: Space after comma:
```suggestion
    >>> plt.plot(x, y)
```

**Review by WarrenWeckesser — CHANGES_REQUESTED** [review]
One more tiny tweak, otherwise looks great!

**Review by WarrenWeckesser — APPROVED** [review]

**Comment by WarrenWeckesser** [comment]
I pushed the one tiny change that I requested.   The one test failure was unrelated to the PR, so I'm merging this.  Thanks @lgfunderburk!

**Comment by lfunderburk** [comment]
Thank you for co-facilitating the SciPy mentored sprint @WarrenWeckesser!

**Pull request merged**